src/UserInterface/templates/driver_index.html

```diff
@@ -160,6 +171,26 @@ <h2>Information</h2>
             }
         };
 
+        socket.on('has_no_nickname', function(data){   
+            if(data.player == '{{ player }}'){
+                document.getElementById("nicknameOverlay").style.display = "flex";
+                save_button = document.getElementById("save_button");
+                save_button.onclick = function(){
+                    if (document.getElementById("textInputNickname").value != "") {
+                        document.getElementById("nicknameOverlay").style.display = "none";
+                        socket.emit('nickname', {player: '{{ player }}', nickname: document.getElementById("textInputNickname").value});
+                    }
+                };
+                document.addEventListener("keydown", function(event) {
+                    if (event.key === "Enter") {
+                        if(document.getElementById("nicknameOverlay").style.display != "none") {
+                            document.getElementById("nicknameOverlay").style.display = "none";
+                            socket.emit('nickname', {player: '{{ player }}', nickname: document.getElementById("textInputNickname").value});
+                        }
```

Comment:
Couldn't this be done via a 'onchange' event listener for the nickname input?

Also it should be checked wether or not the nickname input is empty here as well.

Reply:
Personally, I would prefer to only send the final version of the selected nickname. Also reduces treffic between frontend and backend